File: backend/test/java/de/griefed/serverpackcreator/utilities/commonutilities/StringUtilitiesTest.java
package de.griefed.serverpackcreator.utilities.commonutilities;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class StringUtilitiesTest {

    private final StringUtilities STRINGUTILITIES;

    StringUtilitiesTest() {
        this.STRINGUTILITIES = new StringUtilities();
    }

    @Test
    void buildStringTest() {
        List<String> args = new ArrayList<>(Arrays.asList(
                "config",
                "mods",
                "scripts",
                "seeds",
                "defaultconfigs"
        ));
        String result = STRINGUTILITIES.buildString(args.toString());
        Assertions.assertEquals(args.toString(), String.format("[%s]",result));
    }
}

File: serverpackcreator-api/src/jvmTest/kotlin/de/griefed/serverpackcreator/api/common/StringUtilitiesTest.kt
package de.griefed.serverpackcreator.api.common

import de.griefed.serverpackcreator.api.ApiWrapper
import de.griefed.serverpackcreator.api.utilities.common.StringUtilities
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.io.File

class StringUtilitiesTest internal constructor() {
    private var stringUtilities: StringUtilities =
        ApiWrapper.api(File("src/jvmTest/resources/serverpackcreator.properties")).stringUtilities

    @Test
    fun buildStringTest() {
        val args: List<String> = listOf("config", "mods", "scripts", "seeds", "defaultconfigs")
        val result: String = stringUtilities.buildString(args.toString())
        Assertions.assertEquals(args.toString(), "[%s]".format(result))
    }
}